apache/superset
# feat(trino): add support for user impersonation

### SUMMARY
When using authentication with trino, the current behavior is just to modify the URL to replace the username which will result in an unauthorized exception. This PR will fix this by updating the connection argument with the effective user.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS
Step 1: Setup database connection to trino with credentials in URL.
Step 2: Enable impersonation from `Other` option
Step 3 : Log in with a different user and run the query via SQL lab and you will see the principal user as admin and user as the logged-in user in trino UI.

### ADDITIONAL INFORMATION
